doc/intermediate.rst

```diff
@@ -283,3 +283,36 @@ like this:
 
     ``tqdm`` is not installed by default with Pooch. You will have to install
     it separately in order to use this feature.
+
+Alternatively, you can pass an arbitrary object that behaves like a progress
+that implements the ``update``, ``reset`, and ``close`` methods. ``update``
+should accept a single integer positional argument representing the current
+completion (in bytes), while ``reset`` and ``update`` do not take any argument
+beside ``self``. The object must also have a `total` attribute that can be set
+from outside the class.
+Here's a minimal working example of such a custom "progress display" class
+
+.. code:: python
+
+    import sys
+
+    class MinimalProgressDisplay:
+        def __init__(self, total):
+            self.count = 0
+            self.total = total
+
+        def __repr__(self):
+            return str(self.count) + "/" + str(self.total)
+
+        def render(self):
+            print(f"\r{self}", file=sys.stderr, end="")
+
+        def update(self, i):
+            self.count = i
+            self.render()
+
+        def reset(self):
+            self.count = 0
+
+        def close(self):
+            print("", file=sys.stderr)
```

Comment:
Could you add an example using this class with `HTTPDownloader`?

Reply:
Might also be good to add a ``.. note:`` here saying that this only available for `HTTPDownloader` for now.

Reply:
done !